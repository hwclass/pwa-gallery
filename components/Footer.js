export default class extends React.Component {
  render() {
    return (
      <footer>
        Footer
        <style jsx>{`
          footer {
            grid-area: ft;
            height: 50px;
            background: blue;
            color: #FFF;
            text-align: center;
            line-height: 50px;
          }
        `}</style>
      </footer>
    );
  }
}
