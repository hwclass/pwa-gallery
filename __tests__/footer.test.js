/* eslint-env jest */

import { shallow } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'

import Footer from '../components/Footer.js'

describe('With Enzyme', () => {
  it('Footer shows "Footer"', () => {
    const footer = shallow(<Footer />)

    expect(footer.text()).toEqual('Footer<JSXStyle />')
  })
})

describe('With Snapshot Testing', () => {
  it('Footer shows "Footer"', () => {
    const component = renderer.create(<Footer />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})