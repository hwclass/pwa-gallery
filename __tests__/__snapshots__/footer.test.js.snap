// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Footer shows "Footer" 1`] = `
<footer
  className="jsx-2161472823"
>
  Footer
</footer>
`;
